### ISSUE TITLE
SC 2.4.12 Label in Name feels like the wrong fix to the problem; also applies poorly to non-web ICT

### ISSUE THREAD

**Issue by peterkorn** [w3c/wcag21#762]
As I read the Understanding text, the issue appears to be connected to how products like Dragon Naturally Speaking harvest text via accessibility APIs in order to voice enable web pages / UIs ("Speech input programs also sometimes enable users to speak labels that aren't visible on the page. Speech users can accidentally say a word in a hidden label and be taken to that link without knowing what has happened.").  

This seems to be fundamentally a problem with an insufficiently rich accessibility API, and AT obtaining not the visual text on the screen but the alternative text developed with screen readers in mind, and using it for a different purpose.

I think this needs more study, and more engineering prototyping, to see if we can develop a better & more supported way of voice enabling UIs programmatically, rather than forcing a text generating convention on an accessibility field that was developed for screen readers / TTS output.  I'm also concerned about how this works in non-English, non-Western languages ("best practice is to have the text o the label as the start of the name").  Has that been thought through?

**Comment by awkawk** [comment]
Thank you for commenting. For more information about how the AG WG will process this issue, see the following:
- [Issue Processing](https://www.w3.org/WAI/WCAG21/comments/#issueprocessing) - How the WG will discuss the issue
- [Issue Resolution](https://www.w3.org/WAI/WCAG21/comments/#issueresolution) - Identifying and responding to the official WG response
- [Issue Notification](https://www.w3.org/WAI/WCAG21/comments/#issuenotification) - How to limit notifications from GitHub

**Comment by awkawk** [comment]
@adamsolomon Any response on this coming soon?

**Comment by awkawk** [comment]
[Proposed Response]
Thank you for the comment. The intent of this SC is to recognize that the information presented visually should match or at least contain what is available programmatically. 

The information in the Accessibility API is not solely for use to support vision-related AT, and without specific and separate ways to support AT that are focused on supporting speech users and visually-impaired users the information that is provided needs to serve a broad audience.

If updates to the accessibility APIs offer the possibility of improved guidance we will be able to adjust to that change in Silver.

**Comment by awkawk** [comment]
The WG decided on the above response, so we changed the text in the comment containing the proposed response to read "[Official WG Response]". Please confirm is you are satisfied with the response within 3 days. If we haven't heard a response by then we will regard the resolution as satisfactory.
